### PR TITLE
Tdl 26414 rename custom objects

### PR DESCRIPTION
**Pull request by RushiT0122**
# Description of change
This PR addresses an issue where `custom objects` in HubSpot can have the same names as `standard objects` (e.g., contacts, campaigns), leading to extraction failures [TDL-26380](https://jira.talendforge.org/browse/TDL-26380). 

The fix involves renaming custom objects that share names with standard objects to a distinct `custom_object_<standard_object_name>`. This ensures that custom objects are clearly differentiated from standard objects, preventing conflicts and extraction issues.

# Manual QA steps

- [x] Verify that streams of conflicting custom objects have been renamed correctly.
- [x] Verify that custom objects initially named custom_object_<standard_object_name> are extracted as expected.
- [x] Verify that the standard and renamed custom object streams are extracted properly.

 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

**Comment by sgandhi1311** [comment]
Can you please pull the latest changes from the master branch?

File: tap_hubspot/__init__.py [hunk 1]
```diff
@@ -1422,6 +1426,7 @@ def discover_schemas():
         LOGGER.info('Loading schema for Custom Object - %s', custom_stream["stream"].tap_stream_id)
         result['streams'].append({'stream': custom_stream["stream"].tap_stream_id,
                                   'tap_stream_id': custom_stream["stream"].tap_stream_id,
+                                  "table_name": custom_stream["custom_object_name"],
```
Comment (sgandhi1311) [inline tap_hubspot/__init__.py:1429]: Is it necessary to have new key `table_name`? Because it looks like standard ojects won't be having that key.

File: tests/base.py [hunk 2]
```diff
@@ -170,6 +170,20 @@ def expected_metadata(self):  # DOCS_BUG https://stitchdata.atlassian.net/browse
...
                 self.OBEYS_START_DATE: True
+            },
+            "custom_object_campaigns": {
+                self.PRIMARY_KEYS: {"id"},
+                self.REPLICATION_METHOD: self.INCREMENTAL,
+                self.REPLICATION_KEYS: {"updatedAt"},
+                self.EXPECTED_PAGE_SIZE: 100,
+                self.OBEYS_START_DATE: True
+            },
+            "custom_object_contacts": {
```
Comment (sgandhi1311) [inline tests/base.py:181]: How about testing the corner case scenario for the stream name like - `custom_object_custom_object_contacts`?

File: tap_hubspot/__init__.py (hunk 1 above)
Comment (RushiT0122) [reply tap_hubspot/__init__.py:1429]: Yes, this `table_name` property value will be reflected on the app and in the extracted records. Whereas `stream` will retain the original object name which we need to make API requests.

File: tests/base.py (hunk 2 above)
Comment (RushiT0122) [reply tests/base.py:181]: We only rename custom objects stream if their original name matches with standard stream names like `contacts`, `campaigns`, `deals`. Therefore, if custom object is named as `custom_object_custom_object_contacts` then we won't take any action on it.

**Review by sgandhi1311 — APPROVED** [review]
